components/datetime/src/format/datetime.rs

```diff
@@ -134,9 +132,9 @@ pub(crate) fn write_pattern<'data, T, W, DS, TS>(
     date_symbols: Option<&DS>,
     time_symbols: Option<&TS>,
     loc_datetime: &impl LocalizedDateTimeInput<T>,
-    fixed_decimal_format: &FixedDecimalFormatter,
+    fixed_decimal_format: Option<&FixedDecimalFormatter>,
```

Comment:
Suggestion: the other way to model this, mentioned tersely in the issue, would be to require a `&FixedDecimalFormatter` but use a "last resort" formatter that prints out the desired error syntax. You can build a DecimalSymbolsV1 data struct that has whatever behavior you desire.

Reply:
The FDF is used in a single leaf call afaict, so it should be easy enough to provide a default string there instead of creating a fake FDF.

Reply:
Ok, fine to do it that way, just make sure the output is annotated in some form, since a bare `impl Writeable for u8` doesn't look different from `FixedDecimalFormatter` for many locales.

Reply:
We actually have `impl Writeable for FixedDecimal` which we can use

Reply:
My worry is still that `impl Writeable for FixedDecimal` is too close to `impl Writeable for FormattedFixedDecimal` in most locales so the error might not be noticed. I'd like _something_ that differentiates the number visually.

Reply:
The `Result` will tell the client that there is something wrong with the output. In the current API we can `debug_assert!` the `Result` before throwing it away.

Reply:
I'm debug_asserting in the lossy path now